template.tpl

```diff
@@ -619,86 +683,171 @@ const spDefaultCustomDefs = {
+        value: 'link_url',
       },
       {
         spPropName: 'outbound',
         type: 'boolean',
         ref: 'eventProperty',
-        value: 'outbound'
-      }
-    ]
+        value: 'outbound',
+      },
+    ],
   },
   file_download: {
     eventType: 'ue',
-    eventSchema: 'com.google.ga4.enhanced-measurement/file_download/jsonschema/1-0-0',
-    eventDataMappings : ['file_extention', 'file_name', 'link_classes', 'link_domain', 'link_id', 'link_text', 'link_url']
+    eventSchema:
+      'com.google.ga4.enhanced-measurement/file_download/jsonschema/1-0-0',
+    eventDataMappings: [
+      'file_extention',
+      'file_name',
+      'link_classes',
+      'link_domain',
+      'link_id',
+      'link_text',
+      'link_url',
+    ],
   },
   scroll: {
     eventType: 'ue',
     eventSchema: 'com.google.ga4.enhanced-measurement/scroll/jsonschema/1-0-0',
-    eventDataMappings : []
+    eventDataMappings: [],
   },
   video_complete: {
     eventType: 'ue',
-    eventSchema: 'com.google.ga4.enhanced-measurement/video_complete/jsonschema/1-0-0',
-    eventDataMappings : videoDataMappings
+    eventSchema:
+      'com.google.ga4.enhanced-measurement/video_complete/jsonschema/1-0-0',
+    eventDataMappings: videoDataMappings,
   },
   video_progress: {
     eventType: 'ue',
-    eventSchema: 'com.google.ga4.enhanced-measurement/video_progress/jsonschema/1-0-0',
-    eventDataMappings : videoDataMappings
+    eventSchema:
+      'com.google.ga4.enhanced-measurement/video_progress/jsonschema/1-0-0',
+    eventDataMappings: videoDataMappings,
   },
   video_start: {
     eventType: 'ue',
-    eventSchema: 'com.google.ga4.enhanced-measurement/video_start/jsonschema/1-0-0',
-    eventDataMappings : videoDataMappings
+    eventSchema:
+      'com.google.ga4.enhanced-measurement/video_start/jsonschema/1-0-0',
+    eventDataMappings: videoDataMappings,
   },
   // ga4 automatically collected
   first_visit: {
     eventType: 'ue',
     eventSchema: 'com.google.ga4/first_visit/jsonschema/1-0-0',
-    eventDataMappings : []
+    eventDataMappings: [],
   },
   session_start: {
     eventType: 'ue',
     eventSchema: 'com.google.ga4/session_start/jsonschema/1-0-0',
-    eventDataMappings : []
+    eventDataMappings: [],
   },
   user_engagement: {
     eventType: 'ue',
     eventSchema: 'com.google.ga4/user_engagement/jsonschema/1-0-0',
-    eventDataMappings : ['engagement_time_msec']
+    eventDataMappings: ['engagement_time_msec'],
   },
 };
 
-
 // Helpers
-const fail = (msg) => {
-  log('[ERROR GTM-SS / Snowplow Tag] ' + msg);
+
+/*
+ * Assumes logType argument is string.
+ * Determines if logging is enabled.
+ *
+ * @param logType {string} - the logType set ('no', 'debug', 'always')
+ * @returns - whether logging is enabled (boolean)
+ */
+const determineIsLoggingEnabled = (logType) => {
+  const containerVersion = getContainerVersion();
+  const isDebugMode = containerVersion ? containerVersion.debugMode : false;
```

Comment:
I think adding previewMode check like you mentioned earlier is good idea. As inside the UI we write "debug and preview"